scripts/schema/cleaner.py

```diff
@@ -0,0 +1,186 @@
+# - pre-calculate a few additional helpful fields
+# - converts shorthands into full representation (e.g. reuse locations)
+#
+# This script only deals with field sets themselves and the fields defined
+# inside them. It doesn't perform field reuse, and therefore doesn't
+# deal with final field names either.
+
+
+def clean(fields):
+    visitor.visit_fields(fields, fieldset_func=schema_cleanup, field_func=field_cleanup)
+
+
+# Schema level cleanup
+
+
+def schema_cleanup(schema):
+    # Sanity check first
+    schema_mandatory_attributes(schema)
+    # trailing space cleanup
+    ecs_helpers.dict_clean_string_values(schema['schema_details'])
+    ecs_helpers.dict_clean_string_values(schema['field_details'])
+    # Some defaults
+    schema['schema_details'].setdefault('group', 2)
+    schema['schema_details'].setdefault('root', False)
+    schema['field_details'].setdefault('type', 'group')
+    schema['field_details'].setdefault('short', schema['field_details']['description'])
+    if 'reusable' in schema['schema_details']:
+        # order to perform chained reuses. Set to 1 if it needs to happen earlier.
+        schema['schema_details']['reusable'].setdefault('order', 2)
+    # Precalculate stuff. Those can't be set in the YAML.
+    if schema['schema_details']['root']:
+        schema['schema_details']['prefix'] = ''
+    else:
+        schema['schema_details']['prefix'] = schema['field_details']['name'] + '.'
+    normalize_reuse_notation(schema)
+    # Final validity check
+    schema_assertions_and_warnings(schema)
+
+
+SCHEMA_MANDATORY_ATTRIBUTES = ['name', 'title', 'description']
+
+
+def schema_mandatory_attributes(schema):
+    '''Ensures for the presence of the mandatory schema attributes and raises if any are missing'''
+    current_schema_attributes = sorted(list(schema['field_details'].keys()) +
+                                       list(schema['schema_details'].keys()))
+    missing_attributes = ecs_helpers.list_subtract(SCHEMA_MANDATORY_ATTRIBUTES, current_schema_attributes)
+    if len(missing_attributes) > 0:
+        msg = "Schema {} is missing the following mandatory attributes: {}.\nFound these: {}".format(
+            schema['field_details']['name'], ', '.join(missing_attributes), current_schema_attributes)
+        raise ValueError(msg)
+    if 'reusable' in schema['schema_details']:
+        reuse_attributes = sorted(schema['schema_details']['reusable'].keys())
+        missing_reuse_attributes = ecs_helpers.list_subtract(['expected', 'top_level'], reuse_attributes)
+        if len(missing_reuse_attributes) > 0:
+            msg = "Reusable schema {} is missing the following reuse attributes: {}.\nFound these: {}".format(
+                schema['field_details']['name'], ', '.join(missing_reuse_attributes), reuse_attributes)
+            raise ValueError(msg)
+
+
+def schema_assertions_and_warnings(schema):
+    '''Additional checks on a fleshed out schema'''
+    single_line_short_description(schema)
```

Comment:
It's not really that it will grow, it's that it's being called for schemas and fields :-)

Although it will grow a bit: I do have a note to correct x509's very long `short` description, and add a check to ensure short descriptions are under 100 chars or so.